tests/unit/operators/test_datarobot.py

```diff
@@ -103,31 +105,93 @@ def test_operator_deploy_recommended_model(mocker):
     )
 
 
-def test_operator_score_predictions(mocker):
-    job_mock = mocker.Mock()
-    job_mock.id = "job-id"
-    score_mock = mocker.patch.object(dr.BatchPredictionJob, "score", return_value=job_mock)
-
-    operator = ScorePredictionsOperator(task_id="score_predictions", deployment_id="deployment-id")
-    settings = {
-        "intake_settings": {
+@pytest.fixture(
+    scope="module",
+    params=[
+        {
+            "type": "dataset",
+            "datasetId": "dataset-id",
+        },
+        {
             "type": "s3",
             "url": "s3://path/to/scoring_dataset.csv",
             "credential_id": "credential-id",
         },
+    ],
+)
+def score_settings(request):
+    return {
+        "intake_settings": request.param,
         "output_settings": {
             "type": "s3",
             "url": "s3://path/to/predictions.csv",
             "credential_id": "credential-id",
         },
     }
+
+
+def test_operator_score_predictions(mocker, score_settings):
+    job_id = "job-id"
+
+    # Mocks out the response from the BatchPredictionJob POST request
+    # rather than mocking the whole `score` method
+    post_mock = mocker.Mock()
+    post_mock.json = lambda: job_id
```

Comment:
Can you use `post_mock.json.return_value = job_id` instead? it's easier to read than lambda functions.